src/libcrun/linux.c

```diff
@@ -1947,6 +1947,81 @@ do_mounts (libcrun_container_t *container, int rootfsfd, const char *rootfs, con
   return 0;
 }
 
+/*
+ * libcrun_container_do_bind_mount
+ *
+ *  Allows external plugins and handlers to perform bind `mounts` on container.
+ *  returns: 0 if successful anything else states `error` and configures `err` with relevent error.
+ */
+int
+libcrun_container_do_bind_mount (libcrun_container_t *container, int rootfsfd, const char *rootfs, char *mount_source, char *mount_destination, char **mount_options, size_t mount_options_len, libcrun_error_t *err)
```

Comment:
I think we just need the `container`, the `source` directory, `destination`, `flags` and `data`.

The rootfsfd we can read from container.

Then we safely open `destination` and do a simple `mount`.  It is the caller responsibility to set the other data correctly

Reply:
Sure let me try method.